src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,582 @@
+        .build
+
+      val ehConf = EventHubsConf(connectionString)
+        .setMaxEventsPerTrigger(100)
+        .setStartingPosition(EventPosition.fromStartOfStream)
+
+      val rawEHDF = spark.readStream
+        .format("eventhubs")
+        .options(ehConf.toMap)
+        .load()
+        .withColumn("deserializedBody", 'body.cast("string"))
+      val timestamp = LocalDateTime.now(Pipeline.systemZoneId).toInstant(Pipeline.systemZoneOffset).toEpochMilli
+      val tmpPersistDFPath = s"""$outputPath/${ehName.replaceAll("-", "")}/$timestamp/ehTest/rawDataDF"""
+      val tempPersistChk = s"""$outputPath/${ehName.replaceAll("-", "")}/$timestamp/ehTest/rawChkPoint"""
+
+
+      rawEHDF
+        .writeStream
+        .trigger(Trigger.Once)
+        .option("checkpointLocation", tempPersistChk)
+        .format("delta")
+        .start(tmpPersistDFPath)
+        .awaitTermination()
+
+      val rawBodyLookup = spark.read.format("delta").load(tmpPersistDFPath)
+      val schemaBuilders = rawBodyLookup
+        .withColumn("parsedBody", structFromJson(spark, rawBodyLookup, "deserializedBody"))
+        .select(explode($"parsedBody.records").alias("streamRecord"))
+        .selectExpr("streamRecord.*")
+        .withColumn("version", 'operationVersion)
+        .withColumn("time", 'time.cast("timestamp"))
+        .withColumn("timestamp", unix_timestamp('time) * 1000)
+        .withColumn("date", 'time.cast("date"))
+        .withColumn("organization_id", lit("2222170229861029"))
+        .select('resourceId, 'category, 'version, 'timestamp, 'date, 'properties, 'organization_id, 'identity.alias("userIdentity"))
+        .selectExpr("*", "properties.*").drop("properties")
+
+      val parsedEHDF = rawBodyLookup
+        .withColumn("parsedBody", structFromJson(spark, rawBodyLookup, "deserializedBody"))
+        .select(explode($"parsedBody.records").alias("streamRecord"))
+        .selectExpr("streamRecord.*")
+        .withColumn("version", 'operationVersion)
+        .withColumn("time", 'time.cast("timestamp"))
+        .withColumn("timestamp", unix_timestamp('time) * 1000)
+        .withColumn("date", 'time.cast("date"))
+        .withColumn("organization_id", lit(workspace_id))
+        .select('resourceId, 'category, 'version, 'timestamp, 'date, 'properties, 'organization_id, 'identity.alias("userIdentity"))
+        .withColumn("userIdentity", structFromJson(spark, schemaBuilders, "userIdentity"))
+        .selectExpr("*", "properties.*").drop("properties")
+        .withColumn("requestParams", structFromJson(spark, schemaBuilders, "requestParams"))
+
+      parsedEHDF
+        .drop("response")
+        .verifyMinimumSchema(Schema.auditMasterSchema)
+
+      dbutils.fs.rm(s"""${tmpPersistDFPath}""", true)
+      dbutils.fs.rm(s"""${tempPersistChk}""", true)
+
+      DeploymentValidationReport(true,
+        getSimpleMsg("Validate_EventHub"),
+        testDetails,
+        Some("SUCCESS"),
+        Some(workspace_id)
+      )
+    } catch {
```

Comment:
we need to discuss this validation still -- I don't think it will perform well at large customers